qiskit_ibm_runtime/utils/json.py

```diff
@@ -254,8 +253,14 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
                 serializer=lambda buff, data: dump(data, buff),  # type: ignore[no-untyped-call]
             )
             return {"__type__": "Instruction", "__value__": value}
-        if isinstance(obj, BasePub):
-            return asdict(obj)
+        # TODO proper way to do this?
```

Comment:
I suggest we treat this similarly to other complex data types, e.g. pass a the following dict:

```
        if isinstance(obj, EstimatorPub):
            return {
                "__type__": "EstimatorPub",
                "__value__": {
                    "circuit": obj.circuit,
                    "observables": obj.observables,
                    "parameter_values": obj.parameter_values,
                    "precision": obj.precision,
                },
            }
```

However, I think we should delay that until we start using the `EstimatorPub` from qiskit itself.